### ISSUE TITLE
[Feature Request] sv_hibernate_when_empty

### ISSUE THREAD

**Issue by axl303** [dreamstalker/rehlds#955]
Is it possible to add a function/cvar: sv_hibernate_when_empty 0/1 like CS:GO SRCDS server?
There is no need to spend resources when the servers are empty especially when using -pingboost 3 and 1000FPS.

**Comment by SmileYzn** [comment]
HLDS is suposed to not run any server frame when empty :face_with_spiral_eyes:

**Comment by axl303** [comment]
@justgo97 - Yeah, the idea was to implement it on rehlds, but AMXX plugin can do its job, if ever implemented as you said.

@WaLkZa - Yeah, i saw that before opening the issue, no idea how to implement it efficiently here (poor coding skills).

@SmileYzn https://imgur.com/a/zh0sXZH 
See the graph:
sys_ticrate 1000+ for 1000FPS load && sys_ticrate 100 (hibernating)? :)
The load is from only ONE (1) server, imagine if 10 servers.

**Comment by StevenKal** [comment]
This is totally useless to put "sys_ticrate" to so high value like 1 000, besides, server usually oscillate below (type "stats" from RCON to check live FPS stats as example). Put "sys_ticrate" in the range 250<->300, not more.
For client FPS, enhancing the FPS from 100 to ~200 ("fps_override 1" + "fps_max 200") makes game a bit more fluid, we can feel it, despite human eyes maybe not notice much difference from a specific amount of FPS, but we feel it is more fluid.
But for the server, there is no point to do such, additionnaly to the fact this could lead to some issues about frames or else, as the frequency of calls are increased.

**Comment by abecee** [comment]
> This is totally useless to put "sys_ticrate" to so high value like 1 000, besides, server usually oscillate below (type "stats" from RCON to check live FPS stats as example). Put "sys_ticrate" in the range 250<->300, not more.
> For client FPS, enhancing the FPS from 100 to ~200 ("fps_override 1" + "fps_max 200") makes game a bit more fluid, we can feel it, despite human eyes maybe not notice much difference from a specific amount of FPS, but we feel it is more fluid.
> But for the server, there is no point to do such, additionnaly to the fact this could lead to some issues about frames or else, as the frequency of calls are increased.

Yo. I have been using sys_ticrate above 1000 from server launch and when i type stats from rcon, i get 1000 everytime in every circumstance ( 1 player, 10 players or 14 players ).
The only thing that i will try is with the fps 200 and override 1 to see the difference between fps 100 and override 0 ( mix server by the way )

**Comment by axl303** [comment]
> This is totally useless to put "sys_ticrate" to so high value like 1 000, besides, server usually oscillate below (type "stats" from RCON to check live FPS stats as example). Put "sys_ticrate" in the range 250<->300, not more. For client FPS, enhancing the FPS from 100 to ~200 ("fps_override 1" + "fps_max 200") makes game a bit more fluid, we can feel it, despite human eyes maybe not notice much difference from a specific amount of FPS, but we feel it is more fluid. But for the server, there is no point to do such, additionnaly to the fact this could lead to some issues about frames or else, as the frequency of calls are increased.

Full bullshit. Don't talk shit things. Thank you.

**Comment by SmileYzn** [comment]
I only need to know why persons says that server fps (ticrate) and client fps are related. It is funny 🤣 

One never affects other, sys_ticarate itself is not the problem of cpu usage, but ticrate with pingboost can cause a difference in cpu usage.

[axl303](https://github.com/axl303)
The ticrate Itself can cause CPU usage as i said, but the server is empty of players and bots?
The serverFrame never run when server is empty, this graph do not make sense with a empty server. 
The server you show here is really empty of players and bots?

A small notation of -pingboost parameter, the server gains 2~8 of ping that is not bad, but costs a lot of CPU for nothing.

Finally, HLDS server is not designed to run a lot of instances like in a shared hosting (Do not cry 😆). And will consume CPU if you change sys_ticrate or ping boost values, THAT is the way the server works.

Again, server is not expected to use any CPU if is totally empty.
See AMX task system example, simply not run.

**Comment by axl303** [comment]
Server is totally empty 0 bots 0 players.
1 cpu core = 1 server :) 
Resources are dedicated, not shared. 
Empty or not server is consuming same resources. (!!!!!!!)

Again, server is not expected to use any CPU if is totally empty. - Not true


You fail to notice - empty or not the server is using the same amount cpu. 
The hibernate idea is to put sys_ticrate 100 to minimize the load when empty... Is it hard to understand or?

**Comment by abecee** [comment]
> Server is totally empty 0 bots 0 players.
> 1 cpu core = 1 server :) 
> Resources are dedicated, not shared. 
> Empty or not server is consuming same resources. (!!!!!!!)
> 
> Again, server is not expected to use any CPU if is totally empty. - Not true
> 
> 
> You fail to notice - empty or not the server is using the same amount cpu. 
> The hibernate idea is to put sys_ticrate 100 to minimize the load when empty... Is it hard to understand or? 
> 

Did you try this manually and it's working?

**Comment by axl303** [comment]
Check graph... https://imgur.com/a/zh0sXZH

**Comment by abecee** [comment]
> Check graph... https://imgur.com/a/zh0sXZH

Oh, my bad... personally i'm not interest to do something like this because the servers are not mine, i mean i bought them from a hosting servers but it's a good idea. Try to post a thread on some cs scripting forums to make you a plugin for this.

**Comment by SmileYzn** [comment]
Something is wrong with server, did you have any plugin in server?



Em qui., 2 de mar. de 2023 17:39, axl303 ***@***.***>
escreveu:

**Comment by axl303** [comment]
Nah, It's pure 5v5 server (No addons loaded from metamod) . Thats how it works. The load on the core is fine -pingboost3 n sys_ticrate 1060 for 1000FPS.

When playing 5v5 it is totally normal to rise the ticrate to 1000+, but when empty 0 players, we want to decrease it to 100 for example, so we don't load the server (no need) aka hibernating.

Look how srcds is for example csgo server.

No need to spend resources when servers are empty for example 12 hours a day (just a example).

**Comment by di57inct** [comment]
Imo things as simple as this that can be done with amxx shouldn't be done by the rehlds team. Just make a request on the amxx forums.
